### ISSUE TITLE
Cache the 'get token' calls and/or other repetitive calls.

### ISSUE THREAD

**Issue by seonggbang** [Cimpress-MCP/ruby-mcp-gem#14]
Perhaps it will be worth storing the tokens for clients in a cache to reduce the calls to MCP Auth0 service. 
https://developer.cimpress.io/docs/siboz0XUowgP2K5GkgViQ8aVf8zlOnwF/developers/auth0.html#id5

**Comment by sroylance** [comment]
yes.  We'll need to add a dependency on a JWT library so we can refresh the tokens before they expire.
